src/api/spec/services/meta_controller_service/project_updater_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'rails_helper'
+
+RSpec.describe ::MetaControllerService::ProjectUpdater do
+  let(:project) { create(:project, name: 'openSUSE_41') }
+  let(:admin_user) { create(:admin_user, login: 'Admin') }
+
+  before do
+    User.current = admin_user
+  end
+
+  context 'with valid meta' do
+    let(:meta) do
+      <<-HEREDOC
+      <project name="#{project.name}">
+      <title/><description/>
+      </project>
+      HEREDOC
+    end
+
+    let(:project_updater) { ::MetaControllerService::ProjectUpdater.new(project: project, request_data: Xmlhash.parse(meta)) }
+
+    describe '#new' do
+      it { expect(project_updater).not_to be_nil }
+    end
+
+    describe '#call' do
+      it { expect { project_updater.call }.not_to raise_error }
+    end
+
+    describe '#valid?' do
+      let(:project_updater_call) { project_updater.call }
+      before do
+        project_updater_call
+      end
```

Comment:
That's a bit repetitive. I would drop the `let`. IMO `let`s shouldn't contain method calls anyway (except constructors).

Reply:
> That's a bit repetitive. I would drop  if i can agree with thathe `let`. IMO `let`s shouldn't contain method calls anyway (except constructors).

I'm not sure, looks like a normal pattern to do something like:

```
let(:post_save_meta) { post :update, params: { project_name: 'nonexistent_project' }, xhr: true }
```

which normally for me looks good, and it's a method call. What do you think @bgeuken ? Said that I agree that in this situation is kind of superfluous.

Reply:
> Might happen that a value comes from a method call and thus it's wrapped in a `let`.

if you see, ```call``` returns ```self``` and then i test the instances methods on that.. so i think its inline with what you are saying.

Reply:
nitpick, no @hennevogel ?